test/e2e/nodeportlocal_test.go

```diff
@@ -271,12 +294,13 @@ func NPLTestMultiplePods(t *testing.T) {
 	antreaPod, err := testData.getAntreaPodOnNode(node)
 	r.NoError(err, "Error when getting Antrea Agent Pod on Node '%s'", node)
 
+	targetProtocols := sets.NewString(protocolToString(defaultProtocolTCP))
 	for _, testPodName := range testPods {
-		targetPorts := sets.NewInt(defaultTargetPort)
+		targetPortsProtocols := map[int]sets.String{defaultTargetPort: targetProtocols}
```

Comment:
IMO, the burden should not be on the individual test cases to build this data structure. test cases should have an easy way to define expectations incrementally, building an object that can then be used to validate the final Pod annotation